lib/jxl/enc_image_bundle.cc

```diff
@@ -50,6 +50,33 @@ Status CopyToT(const ImageMetadata* metadata, const ImageBundle* ib,
         // Interleave input.
         if (is_gray) {
           src_buf = rect.ConstPlaneRow(ib->color(), 0, y);
+        } else if (ib->c_current().IsCMYK()) {
+          if (!ib->HasBlack()) {
+            ok.store(false);
+            return;
+          }
+          const float* JXL_RESTRICT row_in0 =
+              rect.ConstPlaneRow(ib->color(), 0, y);
+          const float* JXL_RESTRICT row_in1 =
+              rect.ConstPlaneRow(ib->color(), 1, y);
+          const float* JXL_RESTRICT row_in2 =
+              rect.ConstPlaneRow(ib->color(), 2, y);
+          const float* JXL_RESTRICT row_in3 = rect.ConstRow(ib->black(), y);
+          for (size_t x = 0; x < rect.xsize(); x++) {
+#if JPEGXL_ENABLE_SKCMS
+            // SKCMS does CMYK like JXL: 0 = max ink, 1 = white
+            mutable_src_buf[4 * x + 0] = row_in0[x];
+            mutable_src_buf[4 * x + 1] = row_in1[x];
+            mutable_src_buf[4 * x + 2] = row_in2[x];
+            mutable_src_buf[4 * x + 3] = row_in3[x];
+#else
```

Comment:
This has a similar problem to what #958 fixed. Is there a way that we could have this logic inside of `JxlCms`? Perhaps as a correction pass that only occurs in the LCMS case? It could look like the “Expand from 1 to 3 channels” bit in that other PR for example.

Reply:
OK, I'll move it to the same place where gray gets expanded to 3 channels for SKCMS.

Reply:
Done, PTAL.